reasoning_gym/logic/zebra_puzzles.py

```diff
@@ -0,0 +1,79 @@
+from dataclasses import dataclass
+from random import Random, seed
+from typing import Dict, List, Optional, Tuple
+
+from ..factory import ProceduralDataset, register_dataset
+from .contrib.logic_puzzle.generate import generate_puzzle
+
+
+@dataclass
+class ZebraConfig:
+    """Configuration for zebra puzzle generation"""
+
+    k: int = 4
+    m: int = 4
+    seed: Optional[int] = None
+    size: int = 500
+
+    def validate(self):
+        """Validate configuration parameters"""
+        assert 2 <= self.k <= 7, "k must be between 2 and 7"
+        assert 2 <= self.m <= 7, "m must be between 2 and 7"
+
+
+class ZebraDataset(ProceduralDataset):
+    """Generates Game of Life games with configurable parameters"""
+
+    def __init__(self, config: ZebraConfig):
+        self._prompt_templates = [
+            "What will this Game of Life board look like after {simulation_steps} steps of simulation?\n\n{board}"
+        ]
+
+        super().__init__(config=config, seed=config.seed, size=config.size)
+
+    def __getitem__(self, idx: int) -> dict:
+        """Generate a single Zebra task
+
+        Returns:
+            dict with keys:
+                - question: str, the task description
+                - answer: str, a solution string
+                - metadata: dict with generation parameters
+        """
+        seed(self.seed + idx)
+
+        K = self.config.k
+        M = self.config.m
+        instance, puzzle = generate_puzzle(K, M, "train")
+        q = instance["questions"][0]["question"]
+        a = instance["questions"][0]["answer"]
+        question = str(puzzle) + "\n" + q
+
+        return {
+            "question": question,
+            "answer": a,
+            "metadata": {"K": K, "M": M, "answer": a},
```

Comment:
We probably don't need to replicate answer if representation is identical. (I will add code to automatically add additional metadata for composite generators today, e.g. to track the source dataset which generated the entry...)